hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/QueryCacheEndToEndProvider.java

```diff
@@ -48,36 +49,40 @@ public QueryCacheEndToEndProvider(ContextMutexFactory mutexFactory) {
         this.mapNameToQueryCaches = new ConcurrentHashMap<String, ConcurrentMap<String, InternalQueryCache<K, V>>>();
     }
 
-    public InternalQueryCache<K, V> getOrCreateQueryCache(String mapName, String cacheId,
+    public InternalQueryCache<K, V> getOrCreateQueryCache(String mapName, String cacheName,
                                                           ConstructorFunction<String, InternalQueryCache<K, V>> constructor) {
         ContextMutexFactory.Mutex mutex = mutexFactory.mutexFor(mapName);
         try {
             synchronized (mutex) {
-                ConcurrentMap<String, InternalQueryCache<K, V>> cacheIdToQueryCache
+                ConcurrentMap<String, InternalQueryCache<K, V>> cacheNameToQueryCache
                         = getOrPutIfAbsent(mapNameToQueryCaches, mapName, ctor);
-                InternalQueryCache<K, V> queryCache = cacheIdToQueryCache.get(cacheId);
-                if (queryCache == null) {
-                    queryCache = constructor.createNew(cacheId);
-                    if (queryCache == NULL_QUERY_CACHE) {
-                        queryCache = null;
-                    } else {
-                        cacheIdToQueryCache.put(cacheId, queryCache);
-                    }
+
+                InternalQueryCache<K, V> queryCache = cacheNameToQueryCache.get(cacheName);
+                if (queryCache != null) {
+                    return queryCache;
                 }
-                return queryCache;
+
+                String cacheId = UuidUtil.newUnsecureUuidString();
+                queryCache = constructor.createNew(cacheId);
+                if (queryCache != NULL_QUERY_CACHE) {
+                    cacheNameToQueryCache.put(cacheName, queryCache);
+                    return queryCache;
+                }
+
+                return null;
             }
         } finally {
             closeResource(mutex);
         }
     }
 
-    public void removeSingleQueryCache(String mapName, String cacheId) {
+    public void removeSingleQueryCache(String mapName, String cacheName) {
         ContextMutexFactory.Mutex mutex = mutexFactory.mutexFor(mapName);
         try {
             synchronized (mutex) {
                 Map<String, InternalQueryCache<K, V>> cacheIdToQueryCache = mapNameToQueryCaches.get(mapName);
```

Comment:
Variable should be renamed to `cacheNameToQueryCache` to avoid confusion. There are 2 more such local variables in this class which should be renamed.

Reply:
done, now names should be consistent.